File: app/src/main/java/com/metahumanz/pacilread/model/ChapterRecord.java
package com.metahumanz.pacilread.model;

import org.json.JSONObject;

public class ChapterRecord {
    public long id;
    public long bookId;
    public String title;
    public String bodyHtml;
    public String bodyText;
    public int orderIndex;
    public String bodyTextPath;
    public String bodyTextStorage = "db";
    public long bodyTextSize;

    public JSONObject toJson() {
        JSONObject json = new JSONObject();
        try {
            json.put("id", id);
            json.put("bookId", bookId);
            json.put("title", title != null ? title : "");
            json.put("orderIndex", orderIndex);
            json.put("bodyTextPath", bodyTextPath != null ? bodyTextPath : "");
            json.put("bodyTextStorage", bodyTextStorage != null ? bodyTextStorage : "db");
            json.put("bodyTextSize", bodyTextSize);
        } catch (Exception ignore) {}
        return json;
    }

    public static ChapterRecord fromJson(JSONObject json) {
        ChapterRecord chapter = new ChapterRecord();
        chapter.id = json.optLong("id", 0);
        chapter.bookId = json.optLong("bookId", 0);
        chapter.title = json.optString("title", "");
        chapter.orderIndex = json.optInt("orderIndex", 0);
        chapter.bodyTextPath = json.optString("bodyTextPath", "");
        chapter.bodyTextStorage = json.optString("bodyTextStorage", "db");
        chapter.bodyTextSize = json.optLong("bodyTextSize", 0);
        chapter.bodyHtml = "";
        chapter.bodyText = "";
        return chapter;
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/model/ChapterRecord.kt
package com.metahumanz.pacilread.model

import org.json.JSONObject

class ChapterRecord {
    @JvmField var id: Long = 0
    @JvmField var bookId: Long = 0
    @JvmField var title: String? = null
    @JvmField var bodyHtml: String? = null
    @JvmField var bodyText: String? = null
    @JvmField var orderIndex: Int = 0
    @JvmField var bodyTextPath: String? = null
    @JvmField var bodyTextStorage: String? = "db"
    @JvmField var bodyTextSize: Long = 0

    fun toJson(): JSONObject {
        val json = JSONObject()
        try {
            json.put("id", id)
            json.put("bookId", bookId)
            json.put("title", title ?: "")
            json.put("orderIndex", orderIndex)
            json.put("bodyTextPath", bodyTextPath ?: "")
            json.put("bodyTextStorage", bodyTextStorage ?: "db")
            json.put("bodyTextSize", bodyTextSize)
        } catch (_: Exception) {
        }
        return json
    }

    companion object {
        @JvmStatic
        fun fromJson(json: JSONObject): ChapterRecord = ChapterRecord().apply {
            id = json.optLong("id", 0)
            bookId = json.optLong("bookId", 0)
            title = json.optString("title", "")
            orderIndex = json.optInt("orderIndex", 0)
            bodyTextPath = json.optString("bodyTextPath", "")
            bodyTextStorage = json.optString("bodyTextStorage", "db")
            bodyTextSize = json.optLong("bodyTextSize", 0)
            bodyHtml = ""
            bodyText = ""
        }
    }
}
